tests/resolvers/Mutation/assignUserTag.spec.ts

```diff
@@ -0,0 +1,261 @@
+
+    const spy = vi
+      .spyOn(requestContext, "translate")
+      .mockImplementationOnce((message) => `Translated ${message}`);
+
+    try {
+      const args: MutationAssignUserTagArgs = {
+        input: {
+          userId: adminUser!._id,
+          tagId: Types.ObjectId().toString(),
+        },
+      };
+
+      const context = {
+        userId: adminUser!._id,
+      };
+
+      const { assignUserTag: assignUserTagResolver } = await import(
+        "../../../src/resolvers/Mutation/assignUserTag"
+      );
+
+      await assignUserTagResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(spy).toHaveBeenLastCalledWith(TAG_NOT_FOUND.MESSAGE);
+      expect(error.message).toEqual(`Translated ${TAG_NOT_FOUND.MESSAGE}`);
+    }
+  });
+
+  it(`throws Not Authorized Error if the user is not a superadmin or admin of the organization of the tag beind assigned`, async () => {
+    const { requestContext } = await import("../../../src/libraries");
+
+    const spy = vi
+      .spyOn(requestContext, "translate")
+      .mockImplementationOnce((message) => `Translated ${message}`);
+
+    try {
+      const args: MutationAssignUserTagArgs = {
+        input: {
+          userId: adminUser!._id,
+          tagId: testTag!._id.toString(),
+        },
+      };
+
+      const context = {
+        userId: randomUser!._id,
+      };
+
+      const { assignUserTag: assignUserTagResolver } = await import(
+        "../../../src/resolvers/Mutation/assignUserTag"
+      );
+
+      await assignUserTagResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(error.message).toEqual(
+        `Translated ${USER_NOT_AUTHORIZED_ERROR.MESSAGE}`
+      );
+      expect(spy).toHaveBeenLastCalledWith(
+        `${USER_NOT_AUTHORIZED_ERROR.MESSAGE}`
+      );
+    }
+  });
+
+  it(`throws Error if the user being assigned have not joined the  organization of the tag beind assigned`, async () => {
+    const { requestContext } = await import("../../../src/libraries");
+
+    const spy = vi
+      .spyOn(requestContext, "translate")
+      .mockImplementationOnce((message) => `Translated ${message}`);
+
+    try {
+      const args: MutationAssignUserTagArgs = {
+        input: {
+          userId: randomUser!._id,
+          tagId: testTag!._id.toString(),
+        },
+      };
+
+      const context = {
+        userId: adminUser!._id,
+      };
+
+      const { assignUserTag: assignUserTagResolver } = await import(
+        "../../../src/resolvers/Mutation/assignUserTag"
+      );
+
+      await assignUserTagResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(error.message).toEqual(
+        `Translated ${USER_DOES_NOT_BELONG_TO_TAGS_ORGANIZATION.MESSAGE}`
+      );
+      expect(spy).toHaveBeenLastCalledWith(
+        `${USER_DOES_NOT_BELONG_TO_TAGS_ORGANIZATION.MESSAGE}`
+      );
+    }
+  });
+
+  it(`tag assign should be successful`, async () => {
```

Comment:
`Tag assign should be successful and the user who has been assigned the tag is returned`